online-workflow/src/main/java/eu/itesla_project/online/tools/AmplExportOnlineWorkflowStatesTool.java

```diff
@@ -91,31 +95,57 @@ public void run(CommandLine line) throws Exception {
         OnlineConfig config = OnlineConfig.load();
         OnlineDb onlinedb = config.getOnlineDbFactoryClass().newInstance().create();
         String workflowId = line.getOptionValue("workflow");
+        if (line.hasOption("contingency") && !line.hasOption("state")) {
+            throw new RuntimeException("state needed, when a contingency is specified !");
+        }
+        //pre-contingency state
         List<Integer> states = line.hasOption("state") ? Arrays.asList(Integer.valueOf(line.getOptionValue("state"))) : onlinedb.listStoredStates(workflowId);
+        //post-contingencies states
+        Map<Integer, Set<String>> postContingenciesStates = line.hasOption("contingency")
+                ? ImmutableMap.of(Integer.valueOf(line.getOptionValue("state")), ImmutableSet.of(line.getOptionValue("contingency")))
+                : onlinedb.listStoredPostContingencyStates(workflowId);
+        //total number of post-contingencis states
+        int postcontingenciesStatesSize = postContingenciesStates
+                .keySet()
+                .stream()
+                .flatMap(key -> postContingenciesStates.get(key).stream())
+                .collect(Collectors.toList()).size();
+
         Path folder =  Paths.get(line.getOptionValue("folder"));
-        System.out.println("Exporting in AMPL format network data of workflow " + workflowId + ", " + states.size() + " state(s), to folder " + folder);
-        states.forEach(state -> exportState(onlinedb, workflowId, state, folder));
+        System.out.println("Exporting in AMPL format network data of workflow " + workflowId + ", " + states.size() + " state(s), " +postcontingenciesStatesSize + " post-contingencies states, to folder " + folder);
```

Comment:
Could states can be null (in ListOnlineWorkflowStatesTool.java you do a check).
A space is missing before postcontingenciesStatesSize